SpatialGDK/Source/SpatialGDK/Private/Tests/RPCContainerTest.cpp

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestDefinitions.h"
+
+#include "Utils/RPCContainer.h"
+#include "Schema/RPCPayload.h"
+#include "MockObject.h"
+
+#include "Core.h"
+
+#if WITH_DEV_AUTOMATION_TESTS
+
+using namespace SpatialGDK;
+
+namespace
+{
+	static ESchemaComponentType ReliableType = ESchemaComponentType::SCHEMA_ClientReliableRPC;
+	static ESchemaComponentType UnreliableType = ESchemaComponentType::SCHEMA_ClientUnreliableRPC;
+	static ESchemaComponentType ServerReliableType = ESchemaComponentType::SCHEMA_ServerReliableRPC;
+	static ESchemaComponentType ServerUnreliableType = ESchemaComponentType::SCHEMA_ServerUnreliableRPC;
+	static ESchemaComponentType MulticastType = ESchemaComponentType::SCHEMA_NetMulticastRPC;
+	static ESchemaComponentType CrossServerType = ESchemaComponentType::SCHEMA_CrossServerRPC;
+
+	static FUnrealObjectRef GenerateObjectRef(UObject* TargetObject)
+	{
+		return FUnrealObjectRef{ Worker_EntityId(TargetObject), 0 };
+	}
+
+	static uint32 GeneratePayloadFunctionIndex()
+	{
+		static uint32 freeIndex = 0;
+		return freeIndex++;
+	}
+
+	static FPendingRPCParamsPtr CreateMockParameters(UObject* TargetObject, ESchemaComponentType Type)
+	{
+		// Use PayloadData as a place to store RPC type
+		RPCPayload Payload(0, GeneratePayloadFunctionIndex(), TypeToArray(Type));
+		int ReliableRPCIndex = 0;
+
+		FUnrealObjectRef ObjectRef = GenerateObjectRef(TargetObject);
+
+		return MakeUnique<FPendingRPCParams>(ObjectRef, MoveTemp(Payload), ReliableRPCIndex);
+	}
+}
+
+TEST(FRPCContainer, GIVEN_a_container_WHEN_nothing_has_been_added_THEN_nothing_is_queued)
```

Comment:
If we define a local macro that consumes this TEST macro we can be even more controlled about the use over time. For example

`#define RPCCONTAINER_TEST(TestName) \
  TEST(FRPCContainer, TestName)`

If we change the core TEST macro or if we introduce variants of that macro (eg with different num of parameters) and want to consume them instead then fix up will be more easily controlled.

Reply:
Yes, that should be cleaner and more flexible.

Reply:
Added a new macro as suggested.